### PR TITLE
Add full iris classification example

### PR DESCRIPTION
**Pull request by ruankie**
## Overview

This PR adds the full iris classification example.

## Changes

- Added notebooks for data processing, training, and serving
- Added training pipeline resources
- Added deployment resources
- Added READMEs for all components

### CODE REVIEW

File: iris-classification/training/component_metadata/process_data.yaml
Comment (harryleesan) [inline iris-classification/training/component_metadata/process_data.yaml:1]: @ruankie Do we need this file committed? Isn't this generated by the `kfp build` command?

File: iris-classification/training/component_metadata/train_model.yaml
Comment (harryleesan) [inline iris-classification/training/component_metadata/train_model.yaml:1]: @ruankie Same with this file, do we need to commit this?

**Comment by ruankie** [comment]
Ah good point @harryleesan 
It would probably be less confusing for users if the KFP-generated files are not there. I have removed them and added them to the `.gitignore`.
I'll do the same for other examples in the future too.

File: iris-classification/training/component_metadata/process_data.yaml
Comment (ruankie) [reply iris-classification/training/component_metadata/process_data.yaml:1]: Please see comment below

File: iris-classification/training/component_metadata/train_model.yaml
Comment (ruankie) [reply iris-classification/training/component_metadata/train_model.yaml:1]: Please see comment below

**Review by harryleesan — APPROVED** [review]
LGTM

**Pull request merged**